db/write_batch.cc

```diff
@@ -884,11 +891,11 @@ Status WriteBatchInternal::TimedPut(WriteBatch* b, uint32_t column_family_id,
     PutVarint32(&b->rep_, column_family_id);
   }
   PutLengthPrefixedSlice(&b->rep_, key);
-  PutLengthPrefixedSlice(&b->rep_, value);
-  // For a kTypeValuePreferredSeqno entry, its write time is encoded separately
-  // from value in an encoded WriteBatch. They are packed into one value Slice
-  // once it's written to the database.
-  PutFixed64(&b->rep_, write_unix_time);
+  std::string encoded_write_time;
+  PutFixed64(&encoded_write_time, write_unix_time);
+  std::array<Slice, 2> value_with_time{{value, encoded_write_time}};
+  SliceParts packed_value(value_with_time.data(), 2);
+  PutLengthPrefixedSliceParts(&b->rep_, packed_value);
```

Comment:
I guess I was hoping to get rid of the SliceParts stuff. Here we're fundamentally putting the packed value contiguously in `&b->rep_` so it seems like we can use that for protection info rather than doing a bunch of extra copies for SliceParts stuff.

Reply:
I agree, I started with this hoping that it could save copying to a string buffer, but the `SliceParts` used by the protection info will create such a string buffer anyway, so this is unnecessarily hacky.